### PR TITLE
Xeno armor refactor

### PR DESCRIPTION
**Pull request by Rohesie**
## About The Pull Request

There will be blood (but not yet).

## Why It's Good For The Game

Current armor system must be purged with fire. Armor soak is currently exponential, so low caliber arms are useless and the higher the damage, the lower the soak. It makes shotguns and alpha strike reign unmatched. Also there's no xeno melee armor.

Reworking it is very needed, but balancing it will be hell.

This is a first step towards it, to keep the changes as incremental as possible.

Tried to keep behavior as unchanged as possible for this one.

## Changelog
:cl:
balance: Xeno explosive resistance tweaked a little bit to be more streamlined on the progression of the explosion intensity. Shouldn't be very noticeable.
/:cl:

### CODE REVIEW

File: code/__DEFINES/codex_defines.dm [hunk 1]
```diff
@@ -0,0 +1 @@
+#define ARMOUR_TO_DESCRIPTIVE_TERM ( list("melee" = "melee attacks", "bullet" = "ballistics", "laser" = "lasers", "energy" = "energy", "bomb" = "explosions", "bio" = "biohazards", "rad" = "radiation", "fire" = "fire", "acid" = "acid") )
```
Comment (spookydonut) [inline code/__DEFINES/codex_defines.dm]: use glob list init

File: code/modules/mob/living/carbon/xenomorph/castes/warrior/castedatum_warrior.dm [hunk 2]
```diff
@@ -73,7 +73,7 @@
 	upgrade_threshold = 400
 
 	// *** Defense *** //
-	armor_deflection = 50
+	armor = list("melee" = 5, "bullet" = 5, "laser" = 5, "energy" = 5, "bomb" = 5, "bio" = 5, "rad" = 5, "fire" = 5, "acid" = 5)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/xenomorph/castes/warrior/castedatum_warrior.dm]:
```suggestion
	armor = list("melee" = 50, "bullet" = 50, "laser" = 50, "energy" = 50, "bomb" = 50, "bio" = 50, "rad" = 50, "fire" = 50, "acid" = 50)
```

File: code/modules/mob/living/carbon/xenomorph/castes/crusher/castedatum_crusher.dm [hunk 3]
```diff
@@ -136,5 +136,5 @@
 	deevolves_to = /mob/living/carbon/xenomorph/warrior
 
 	// *** Defense *** //
-	armor_deflection = 100
+	armor = list("melee" = 10, "bullet" = 10, "laser" = 10, "energy" = 10, "bomb" = 10, "bio" = 10, "rad" = 10, "fire" = 10, "acid" = 10)
```
Comment (InterroLouis) [inline code/modules/mob/living/carbon/xenomorph/castes/crusher/castedatum_crusher.dm]: armor = list("melee" = 100, "bullet" = 100, "laser" = 100, "energy" = 100, "bomb" = 100, "bio" = 100, "rad" = 100, "fire" = 100, "acid" = 100)

File: code/modules/mob/living/carbon/xenomorph/castes/queen/castedatum_queen.dm [hunk 4]
```diff
@@ -76,7 +76,7 @@
 	upgrade_threshold = 1600
 
 	// *** Defense *** //
-	armor_deflection = 50
+	armor = list("melee" = 5, "bullet" = 5, "laser" = 5, "energy" = 5, "bomb" = 5, "bio" = 5, "rad" = 5, "fire" = 5, "acid" = 5)
```
Comment (InterroLouis) [inline code/modules/mob/living/carbon/xenomorph/castes/queen/castedatum_queen.dm]: armor = list("melee" = 50, "bullet" = 50, "laser" = 50, "energy" = 50, "bomb" = 50, "bio" = 50, "rad" = 50, "fire" = 50, "acid" = 50)

File: code/modules/mob/living/carbon/xenomorph/castes/queen/castedatum_queen.dm [hunk 5]
```diff
@@ -113,7 +113,7 @@
 	upgrade_threshold = 3200
 
 	// *** Defense *** //
-	armor_deflection = 55
+	armor = list("melee" = 5, "bullet" = 5, "laser" = 5, "energy" = 5, "bomb" = 5, "bio" = 5, "rad" = 5, "fire" = 5, "acid" = 5)
```
Comment (InterroLouis) [inline code/modules/mob/living/carbon/xenomorph/castes/queen/castedatum_queen.dm]: armor = list("melee" = 55, "bullet" = 55, "laser" = 55, "energy" = 55, "bomb" = 55, "bio" = 55, "rad" = 55, "fire" = 55, "acid" = 55)

File: code/modules/mob/living/carbon/xenomorph/castes/warrior/castedatum_warrior.dm (hunk 2 above)
Comment (InterroLouis) [inline code/modules/mob/living/carbon/xenomorph/castes/warrior/castedatum_warrior.dm]: armor = list("melee" = 50, "bullet" = 50, "laser" = 50, "energy" = 50, "bomb" = 50, "bio" = 50, "rad" = 50, "fire" = 50, "acid" = 50)

**Comment by spookydonut** [comment]
Good first attempt needs more discussion about better code design

File: code/modules/mob/living/carbon/xenomorph/castes/warrior/castedatum_warrior.dm [hunk 6]
```diff
@@ -104,7 +104,7 @@
 	upgrade_threshold = 800
 
 	// *** Defense *** //
-	armor_deflection = 50
+	armor = list("melee" = 5, "bullet" = 5, "laser" = 5, "energy" = 5, "bomb" = 5, "bio" = 5, "rad" = 5, "fire" = 5, "acid" = 5)
```
Comment (InterroLouis) [inline code/modules/mob/living/carbon/xenomorph/castes/warrior/castedatum_warrior.dm]: armor = list("melee" = 50, "bullet" = 50, "laser" = 50, "energy" = 50, "bomb" = 50, "bio" = 50, "rad" = 50, "fire" = 50, "acid" = 50)

File: code/modules/mob/living/carbon/xenomorph/castes/warrior/castedatum_warrior.dm [hunk 7]
```diff
@@ -136,7 +136,7 @@
 	upgrade_threshold = 800
 
 	// *** Defense *** //
-	armor_deflection = 55
+	armor = list("melee" = 5, "bullet" = 5, "laser" = 5, "energy" = 5, "bomb" = 5, "bio" = 5, "rad" = 5, "fire" = 5, "acid" = 5)
```
Comment (InterroLouis) [inline code/modules/mob/living/carbon/xenomorph/castes/warrior/castedatum_warrior.dm]: armor = list("melee" = 55, "bullet" = 55, "laser" = 55, "energy" = 55, "bomb" = 55, "bio" = 55, "rad" = 55, "fire" = 55, "acid" = 55)

**Comment by DominikPanic** [comment]
@InterroLouis currently xeno armors are not 1:1 with human armors so it's not possible to just directly steal the values

**Comment by InterroLouis** [comment]
@DominikPanic It's typo fixes. The progression before was wrong because of some left out zeros, and fives., so how it was was having those castes suddenly become tissue paper at certain upgrade levels.

**Comment by Rohesie** [comment]
This update is less exciting than initially announced because this will be a big rework and it's better to atomize it.
For now it just makes xenos use the armor datum, which will allow us to separate their different resistances more properly.

**Comment by Rohesie** [comment]
Need to investigate if one runtime is caused by this.

**Comment by Rohesie** [comment]
Runtime fixed.

**Comment by Rohesie** [comment]
This one is ready and was tested.

**Pull request merged**